docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+    <p>We can sign this JWT using the JWTSigner we created in step 2:</p>
+    <pre><code class="language-swift">let signedJWT = try myJWT.sign(using: App.jwtSigner)</code></pre>
+    <p>Finally we return the signed JWT String to the User:</p>
+    <pre><code class="language-swift">response.send(signedJWT)</code></pre>
+    <p>Our completed login route should look as follows:</p>
+    <pre><code class="language-swift">app.router.post("/jwtlogin") { request, response, next in
+    let credentials = try request.read(as: UserCredentials.self)
+    // Users credentials are authenticated
+    let myClaims = ClaimsStandardJWT(iss: "Kitura", sub: credentials.username, exp: Date(timeIntervalSinceNow: 3600))
+    var myJWT = JWT(claims: myClaims)
+    let signedJWT = try myJWT.sign(using: App.signer)
+    response.send(signedJWT)
+    next()
+}</code></pre>
+<div class="underline"></div>
+
+<h2 class="heading-2"><span class="blue-text">Step 6:</span> Test the JWT creation</h2>
+<p>Compile your project and start the server.</p>
+<p>To test the route using curl, open Terminal and enter the following:</p>
+<pre><code>curl -X POST \
+http://localhost:8080/jwtlogin \
+-H 'content-type: application/json' \
+-d '{
+"username": "Joe Bloggs",
+"password": "password"
+}'</code></pre>
+<p>You should be returned a JWT String that is structured `xxxx.yyyy.zzzz`
+     where xxxx is the base64 encoded header, yyyy is the base 64 encoded claims and zzzz is the signature.</p>
+<p>The following is a JWT generated using hs256 and the password "kitura":<p>
+<pre><code>eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJLaXR1cmEiLCJzdWIiOiJKb2UgQmxvZ2dzIiwiZXhwIjoxNTUzMDE4Mjg0LjMyOTcwMTl9.t55WealACtYGCQGS3EQgRQuurmNSBO5fWZqzqJjEIi</code></pre>
+<p>You can decode your JWT string to view the headers and claims at <a href="https://jwt.io/#debugger" rel="noopener" target="_blank">JWT.io</a></p>
+    <div class="underline"></div>
+
+    <h2 class="heading-2"><span class="blue-text">Step 7:</span> Verify a JWT</h2>
+    <p class="block-text">We have created a signed JWT for a user to authenticate themselves.
+         At this stage, the user would attach the JWT string to future requests either using cookies or the Authorization header.
+     When we recieve this JWT string on other routes we need to verify that we signed it and it hasn't been altered.</p>
+     <p>Lets start by reading the JWT string from the Authorization header:</p>
+     <p>Inside the get route, add the following code:</p>
+     <pre><code class="language-swift">let authHeader = request.headers["Authorization"]
+guard let authComponents = authHeader?.components(separatedBy: " "),
+    authComponents.count == 2,
+    authComponents[0] == "Bearer"
+else {
+    let _ = response.send(status: .unauthorized)
+    return try response.end()
+}</code></pre>
+    <p>Our JWT String will be `authComponents[1]. We will verify this string and use it to inialize a JWT.</p>
```

Comment:
"The JWT string will be in `authComponents[1]`, so we need to verify this string and use it to initialize a JWT."